databuilder/query_language.py

```diff
@@ -374,3 +374,25 @@ def build_event_table(name, schema, contract=None):
     return EventFrame(
         qm.SelectTable(name, schema=qm.TableSchema(schema)),
     )
+
+
+# CASE EXPRESSION FUNCTIONS
+#
+
+
+class when:
+    def __init__(self, condition):
+        self._condition = condition
+
+    def then(self, value):
+        new = self.__class__(self._condition)
+        new._value = value
+        return new
+
+
+def case(*when_thens, default=None):
+    cases = _DictArg((case._condition, case._value) for case in when_thens)
+    # If we don't want a default then we shouldn't supply an argument, or else it will
+    # get converted into `Value(None)` which is not what we want
+    default_arg = [default] if default is not None else []
+    return _apply(qm.Case, cases, *default_arg)
```

Comment:
I'd be tempted to make this really explicit with an `if` statement.

Reply:
Or even:

```
if default is None:
    return _apply(qm.Case, cases)
else:
    return _apply(qm.Case, cases, default)
```

?

Reply:
Ha, oh yes! For some reason I was blind to the possibility of writing the function call twice. That's much simpler in this case.